### ISSUE TITLE
[C++ API] The cluster_mode_test.cc fails if it was made run functions that have no arguments. 

### ISSUE THREAD

**Issue by AmeerHajAli** [ray-project/ray#11336]
The cluster_mode_test.cc fails if it was made run functions that have no arguments. 
e.g.,
```
int func() { return 1; }

  auto task_obj = Ray::Task(func).Remote();
  int task_result = *(Ray::Get(task_obj));
  EXPECT_EQ(1, task_result);
```

The output error:
```
E1013 08:11:39.012250  2800  2917 core_worker.cc:419] Will resubmit task after a 5000ms delay: Type=NORMAL_TASK, Language=CPP, Resources: {}, function_descriptor={type=CppFunctionDescriptor, lib_name=, function_offset=961856, exec_function_offset=982992}, task_id=df5a1a828c9685d3ffffffff01000000, task_name=+961856, job_id=01000000, num_args=1, num_returns=1
E1013 08:11:45.001149  2800  2917 task_manager.cc:323] Task failed: IOError: 14: Socket closed: Type=NORMAL_TASK, Language=CPP, Resources: {}, function_descriptor={type=CppFunctionDescriptor, lib_name=, function_offset=961856, exec_function_offset=982992}, task_id=df5a1a828c9685d3ffffffff01000000, task_name=+961856, job_id=01000000, num_args=1, num_returns=1
```
- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/latest/installation.html).

**Comment by AmeerHajAli** [comment]
Hi @SongGuyang, can you please look into this issue?

**Comment by SongGuyang** [comment]
Thanks, I will see this.

**Comment by AmeerHajAli** [comment]
Fixed in #11363.
